util/logtools/arg_parser.ts

```diff
@@ -0,0 +1,88 @@
+import argparse from 'argparse';
+
+type argsObject = {
+  [index: string]: string | number | boolean | undefined;
+  'file'?: string;
+  'force'?: boolean;
+  'search_fights'?: number;
+  'search_zones'?: number;
+  'fight_regex'?: string;
+  'zone_regex'?: string;
+  'adjust'?: number;
+};
+
+class TimelineParse {
+  parser = new argparse.ArgumentParser({
+    addHelp: true,
+  });
+  // At least one non-file argument must be selected from this group.
+  fileGroup = this.parser.addArgumentGroup({
+    title: 'File Args',
+    description: 'A file and/or a timeline must be selected before performing operations',
+  });
+  // Exactly one argument should be selected from this group.
+  requiredGroup = this.parser.addMutuallyExclusiveGroup({ required: true });
+  args: argsObject;
+  constructor() {
+    this.fileGroup.addArgument(['-f', '--file'], {
+      help: 'Network log file to analyze',
+    });
+    this.fileGroup.addArgument(['-t', '--timeline'], {
+      help: 'The filename of the timeline to test against, e.g. ultima_weapon_ultimate',
+    });
+    this.parser.addArgument(['--force'], {
+      nargs: 0,
+      help: 'Overwrite files when exporting',
+    });
+    this.requiredGroup.addArgument(['-lf', '--search-fights'], {
+      nargs: '?',
+      constant: -1,
```

Comment:
Is it possible to not have magic constants here? Could this be `null` or even better `undefined`?  I'm not sure I understand why you need to check for both `-1` and `null` below.  Could you explain?

Reply:
I'm not completely sure whether it's possible. We need a constant defined in order for bare flags to work *at all*, and the Typescript version of `argparse` treats falsy constants as if the bare flags weren't actually present. (That is, if `--foo` has no constant definition, or if that constant is falsy, there is no difference between `[invoke]` and `[invoke] --foo`.)

That being said, you're entirely correct that if the user is not printing the list of encounters/zones, we could simplify the subsequent check down to `if(args[arg])`